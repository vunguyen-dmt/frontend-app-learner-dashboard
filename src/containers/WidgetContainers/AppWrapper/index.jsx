import React from 'react';
import PropTypes from 'prop-types';

import PaintedDoorExperimentProvider from 'widgets/RecommendationsPaintedDoorBtn/PaintedDoorExperimentContext';

export const AppWrapper = ({
  children,
}) => {
  if (process.env.EXPERIMENT_08_23_VAN_PAINTED_DOOR) {
    return (
      <PaintedDoorExperimentProvider>
        {children}
      </PaintedDoorExperimentProvider>
    );
  }
  return children;
};
AppWrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
};

export default AppWrapper;
