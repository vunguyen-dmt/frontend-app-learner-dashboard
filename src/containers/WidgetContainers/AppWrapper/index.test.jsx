import React from 'react';

import { shallow } from '@edx/react-unit-test-utils';

import PaintedDoorExperimentProvider from 'widgets/RecommendationsPaintedDoorBtn/PaintedDoorExperimentContext';

import AppWrapper from '.';

jest.mock(
  'widgets/RecommendationsPaintedDoorBtn/PaintedDoorExperimentContext',
  () => 'PaintedDoorExperimentProvider',
);

let el;

const children = (<div>This is some <b>test</b> <i>content</i></div>);

const render = () => {
  el = shallow(<AppWrapper>{children}</AppWrapper>);
};

const mockAndRenderForBlock = (newVal) => {
  const oldVal = process.env;
  beforeEach(() => {
    process.env = { ...oldVal, ...newVal };
    render();
  });
  afterEach(() => {
    process.env = oldVal;
    render();
  });
};

describe('AppWrapper WidgetContainer component', () => {
  describe('output', () => {
    describe('painted door experiment is active (08/23)', () => {
      mockAndRenderForBlock({ EXPERIMENT_08_23_VAN_PAINTED_DOOR: true });
      test('snapshot', () => {
        expect(el.snapshot).toMatchSnapshot();
      });
      it('renders children wrapped in PaintedDoorExperimentProvider', () => {
        const control = el.instance.findByType(PaintedDoorExperimentProvider)[0];
        expect(el.instance).toEqual(control);
      });
    });
    describe('no experiments are active', () => {
      mockAndRenderForBlock({ EXPERIMENT_08_23_VAN_PAINTED_DOOR: false });
      test('snapshot', () => {
        expect(el.snapshot).toMatchSnapshot();
      });
      it('renders children wrapped in PaintedDoorExperimentProvider', () => {
        expect(el.instance.matches(shallow(children))).toEqual(true);
      });
    });
  });
});
