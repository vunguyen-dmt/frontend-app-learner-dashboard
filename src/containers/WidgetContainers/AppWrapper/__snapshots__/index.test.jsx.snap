// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppWrapper WidgetContainer component output no experiments are active snapshot 1`] = `
<div>
  This is some 
  <b>
    test
  </b>
   
  <i>
    content
  </i>
</div>
`;

exports[`AppWrapper WidgetContainer component output painted door experiment is active (08/23) snapshot 1`] = `
<PaintedDoorExperimentProvider>
  <div>
    This is some 
    <b>
      test
    </b>
     
    <i>
      content
    </i>
  </div>
</PaintedDoorExperimentProvider>
`;
